### PR TITLE
Adding custom shader support to the engine.

### PR DESCRIPTION
**Pull request by Secticide**
My colleges and I have been working on a drawing application using the Flambe engine and have found it difficult to create custom effects and such. The solution to this was the add additional functionality to the engine to allow for custom shaders.

On the WebGL side; everything is complete and works correctly, however the HXSL stuff is a little more difficult to get working.

I was wondering if this feature would be welcome within the engine in the near future?

There is a great deal more testing to do (especially ensuring the system is backwards-compatible) all of which I plan to do as soon as I get the system in place and fully functional.

I've also been writing a wiki-style entry on how this system will be used (if it gets added to the engine). Please look over my additions/changes and comment on anything that may need to be changed/improved.

Thanks,
Jak Boulton.

### CODE REVIEW

File: src/flambe/platform/Effect.hx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Effect, an abstraction of the platform specific shaders
+ */
+interface Effect
+{
+#if flash
+    public var shader:Shader;
+
+    public function instansiate() :Void;
```
Comment (markknol) [inline src/flambe/platform/Effect.hx]: Shouldn't this be `instantiate`?

Comment (Secticide) [inline src/flambe/platform/Effect.hx]: Yep. My bad, will commit the changes shortly.

**Comment by Secticide** [comment]
Thanks for the quick comments Mark.

**Comment by Secticide** [comment]
When writing a group of demos displaying a collection of potential uses for custom shaders; I've come to the realisation that a great use for this new custom shader system would be to have post-process effects defined by these shaders. I'll look into writing a new component, or maybe even having this as an addition to the renderer since it's not an object-specific effect. Any additional comments or ideas are welcome!

**Pull request closed without merging**